packages/tanstack-start/src/client/uiComponents.tsx

```diff
@@ -1,10 +1,82 @@
-export {
-  SignIn,
-  SignUp,
-  UserButton,
-  UserProfile,
-  OrganizationProfile,
-  OrganizationSwitcher,
-  OrganizationList,
-  CreateOrganization,
+import {
+  CreateOrganization as BaseCreateOrganization,
+  OrganizationList as BaseOrganizationList,
+  OrganizationProfile as BaseOrganizationProfile,
+  SignIn as BaseSignIn,
+  SignUp as BaseSignUp,
+  UserProfile as BaseUserProfile,
 } from '@clerk/clerk-react';
+import { useRoutingProps } from '@clerk/clerk-react/internal';
+import type {
+  CreateOrganizationProps,
+  OrganizationProfileProps,
+  SignInProps,
+  SignUpProps,
+  UserProfileProps,
+} from '@clerk/types';
+import { useLocation, useParams } from '@tanstack/react-router';
+
+const usePathnameWithoutSplatRouteParams = () => {
```

Comment:
I can make that happen in a separate PR as this is almost the same function as we have in Remix ([ref](https://github.com/clerk/javascript/blob/main/packages/remix/src/client/usePathnameWithoutSplatRouteParams.tsx)), so it can be extracted and be put in the `@clerk/shared` to be used by both, and I will also add tests then